src/public/modules/forms/helpers/CsvMergedHeadersGenerator.ts

```diff
@@ -1,21 +1,38 @@
-const moment = require('moment-timezone')
-const keyBy = require('lodash/keyBy')
-const { CsvGenerator } = require('./CsvGenerator')
-const { getResponseInstance } = require('./response-factory')
-
-/**
- * @typedef {{
- *    _id: string,
- *    question: string,
- *    answer?: string,
- *    answerArray?: string[],
- *    fieldType: string,
- *    isHeader?: boolean,
- * }} DisplayedResponse
- */
-
-class CsvMergedHeadersGenerator extends CsvGenerator {
-  constructor(expectedNumberOfRecords, numOfMetaDataRows) {
+import keyBy from 'lodash/keyBy'
+import moment from 'moment-timezone'
+
+import { DisplayedResponseWithoutAnswer } from '../../../../types/response'
+
+import { Response } from './csv-response-classes'
+import { CsvGenerator } from './CsvGenerator'
+import { getResponseInstance } from './response-factory'
+
+type UnprocessedRecord = {
+  created: string
+  submissionId: string
+  record: { [fieldId: string]: Response }
```

Comment:
noticed the `{ [fieldId: string]: Response }` shape is repeated in a couple of places. we can extract that into a type for clarity